testutil/integration/network/setup.go

```diff
@@ -0,0 +1,211 @@
+	for i := 0; i < numberOfValidators; i++ {
+		privVal := mock.NewPV()
+		pubKey, _ := privVal.GetPubKey()
+		validator := tmtypes.NewValidator(pubKey, 1)
+		tmValidators[i] = validator
+	}
+
+	return tmtypes.NewValidatorSet(tmValidators)
+}
+
+// fundAndCreateGenesisAccounts funds accounts
+// create genesis accounts
+func createGenesisAccounts(accounts []sdktypes.AccAddress) []authtypes.GenesisAccount {
+	numberOfAccounts := len(accounts)
+	genAccounts := make([]authtypes.GenesisAccount, numberOfAccounts)
+	for i, acc := range accounts {
+		baseAcc := authtypes.NewBaseAccount(acc, nil, 0, 0)
+		genAccounts[i] = baseAcc
+	}
+	return genAccounts
+}
+
+func createBalances(accounts []sdktypes.AccAddress, coin sdktypes.Coin) []banktypes.Balance {
+	numberOfAccounts := len(accounts)
+	fundedAccountBalances := make([]banktypes.Balance, numberOfAccounts)
+	for i, acc := range accounts {
+		balance := banktypes.Balance{
+			Address: acc.String(),
+			Coins:   sdktypes.NewCoins(coin),
+		}
+
+		fundedAccountBalances[i] = balance
+	}
+	return fundedAccountBalances
+}
+
+func createEvmosApp(chainID string) *app.Evmos {
+	// Create evmos app
+	db := dbm.NewMemDB()
+	logger := log.NewNopLogger()
+	loadLastest := true
+	skipUpgradeHeights := map[int64]bool{}
+	homePath := app.DefaultNodeHome
+	invCheckPeriod := uint(5)
+	encodingConfig := encoding.MakeConfig(app.ModuleBasics)
+	appOptions := simutils.NewAppOptionsWithFlagHome(app.DefaultNodeHome)
+	baseAppOptions := []func(*baseapp.BaseApp){baseapp.SetChainID(chainID)}
+
+	return app.NewEvmos(
+		logger,
+		db,
+		nil,
+		loadLastest,
+		skipUpgradeHeights,
+		homePath,
+		invCheckPeriod,
+		encodingConfig,
+		appOptions,
+		baseAppOptions...,
+	)
+}
+
+func createStakingValidator(val *tmtypes.Validator, amtOfBondedTokens sdkmath.Int) (stakingtypes.Validator, error) {
+	pk, err := cryptocodec.FromTmPubKeyInterface(val.PubKey)
+	if err != nil {
+		return stakingtypes.Validator{}, err
+	}
+
+	pkAny, err := codectypes.NewAnyWithValue(pk)
+	if err != nil {
+		return stakingtypes.Validator{}, err
+	}
+
+	commission := stakingtypes.NewCommission(sdktypes.ZeroDec(), sdktypes.ZeroDec(), sdktypes.ZeroDec())
+	validator := stakingtypes.Validator{
+		OperatorAddress:   sdktypes.ValAddress(val.Address).String(),
+		ConsensusPubkey:   pkAny,
+		Jailed:            false,
+		Status:            stakingtypes.Bonded,
+		Tokens:            amtOfBondedTokens,
+		DelegatorShares:   sdktypes.OneDec(),
+		Description:       stakingtypes.Description{},
+		UnbondingHeight:   int64(0),
+		UnbondingTime:     time.Unix(0, 0).UTC(),
+		Commission:        commission,
+		MinSelfDelegation: sdktypes.ZeroInt(),
+	}
+	return validator, nil
+}
+
+func createStakingValidators(tmValidators []*tmtypes.Validator, bondedAmt sdkmath.Int) ([]stakingtypes.Validator, error) {
+	amountOfValidators := len(tmValidators)
+	stakingValidators := make([]stakingtypes.Validator, 0, amountOfValidators)
```

Comment:
some of the utils are using this zero allocation declaration while others aren't, best align those